libres/lib/analysis/ies/ies_enkf.cpp

```diff
@@ -372,46 +366,58 @@ void ies_enkf_updateA(
 *  Additionally we may loose realizations during the iteratitions, and we extract measurement perturbations for
 *  only the active realizations.
 */
-void ies_enkf_linalg_extract_active(const ies_enkf_data_type *data,
-                                    matrix_type *E, FILE *log_fp, bool dbg) {
-
-    const bool_vector_type *obs_mask = ies_enkf_data_get_obs_mask(data);
-    const bool_vector_type *ens_mask = ies_enkf_data_get_ens_mask(data);
 
-    int obs_size_msk =
-        ies_enkf_data_get_obs_mask_size(data); // Total number of observations
-    int ens_size_msk =
-        ies_enkf_data_get_ens_mask_size(data); // Total number of realizations
-
-    const matrix_type *dataE = ies_enkf_data_getE(data);
-
-    int m = -1; // counter for currently active measurements
-    for (int iobs = 0; iobs < obs_size_msk; iobs++) {
-        if (bool_vector_iget(obs_mask, iobs)) {
-            m++;
-            fprintf(log_fp, "ies_enkf_linalg_extract_active iobs=%d m=%d)\n",
-                    iobs, m);
-            int i = -1;
-            for (int iens = 0; iens < ens_size_msk; iens++) {
-                if (bool_vector_iget(ens_mask, iens)) {
-                    i++;
-                    matrix_iset_safe(E, m, i, matrix_iget(dataE, iobs, iens));
+static matrix_type *alloc_active(const matrix_type *full_matrix,
```

Comment:
Could you also change the documentation for this function, it seems oddly specific to ies now, for what the function actually does.

Reply:
Well it is `static` - of course it is specific to the `ies` implementation.

Reply:
Sorry, I was ambigous, I was thinking the documentation above seems specific to ies, while the function is much more general.

Reply:
OK - the function in question is an implementation detail for the three ies specific functions:
```C++
matrix_type *ies_enkf_alloc_activeE();
matrix_type *ies_enkf_alloc_activeW();
matrix_type *ies_enkf_alloc_activeA();
```
I'll shuffle things around so it hopefully becomes clearer.